js/src/reader/arrow.ts

```diff
@@ -52,9 +52,12 @@ export function* readBuffers(...bytes: Array<Uint8Array | Buffer | string>) {
         let index = -1, fieldsLength = schema.fieldsLength();
         if (batch.id) {
             while (++index < fieldsLength) {
+                let found: boolean = false;
                 for (let [id, vector] of readDictionaries(schema.fields(index), batch, state, dictionaries)) {
                     dictionaries[id] = dictionaries[id] && dictionaries[id].concat(vector) || vector;
+                    found = true;
```

Comment:
@TheNeuralBit I think we can `break` here? IIRC the `readDictionaries` function [returns](https://github.com/apache/arrow/blob/master/js/src/reader/dictionary.ts#L36) as soon as it finds a dictionary, or yields nothing if it doesn't

Reply:
but breaking here would only break out of the `for`, not the `while`, right?
it does seem like there should be a cleaner way, since `readDictionaries` will return at most one vector. Is it necessary to return the `id`? shouldn't it always equal `batch.id`?
Maybe `readDictionaries` could just return 'vector' if found, otherwise null

Reply:
@TheNeuralBit yeah, that's right. I think both the `for` loop over the `readDictionaries` iterator and the return `[id, vector]` tuple are remnants from when my understanding was that DictionaryBatches could contain more than one DictionaryVector.

Since that was a misunderstanding, it totally makes sense to change the `readDictionaries` signature to return `DictionaryVector | null` instead of `Iterator<[id, vector]>`, and not loop. Do you mind updating the PR to reflect this?

Reply:
sure!